tests/integration/kafka_test.go

```diff
@@ -295,6 +295,24 @@ func TestKafkaIntegration_bootstrap_localOnlyCollection(t *testing.T) {
 	}
 }
 
+func TestKafkaIntegration_bootstrap_localOnlyCollectionEnforcingTopicCollection(t *testing.T) {
+	bootstrapDiscoverConfigLocalOnlyCollection := func(command []string) []string {
+		return append(bootstrapDiscoverConfig(command), "--local_only_collection", "--force_topic_sample_collection")
+	}
+
+	stdout, stderr, err := runIntegration(t, bootstrapDiscoverConfigLocalOnlyCollection)
+
+	assert.NotNil(t, stderr, "unexpected stderr")
+	assert.NoError(t, err, "Unexpected error")
+
+	schemaPath := filepath.Join("json-schema-files", "kafka-schema-only-local-force-topic-collection.json")
+	err = jsonschema.Validate(schemaPath, stdout)
+	assert.NoError(t, err, "The output of kafka integration doesn't have expected format.")
+	for _, topic := range topicNames {
+		assert.Contains(t, stdout, topic, fmt.Sprintf("The output doesn't have the topic %s", topic))
+	}
```

Comment:
Did you consider adding an additional test checking that the TopicSample is not there when `--force_topic_sample_collection` is not set?

Reply:
Maybe it doesn't make sense because it can cause data duplication.

Reply:
there is already the old test covering that!